src/platform_impl/android/mod.rs

```diff
@@ -988,70 +983,169 @@ impl Ord for MonitorHandle {
 }
 
 impl MonitorHandle {
-    pub(crate) fn new(app: AndroidApp) -> Self {
-        Self { app }
+    fn primary_monitor(app: AndroidApp) -> Option<Self> {
+        let jvm = ffi::jvm(&app);
+        let mut env = ffi::env(&jvm);
+
+        let default_display = ffi::Display::default_display(&mut env);
+        let id = ffi::context(&mut env)
+            .display_manager(&mut env)
+            .display(&mut env, default_display)
+            ?
+            .display_id(&mut env);
+
+        Some(Self { app, id })
     }
 
-    pub fn name(&self) -> Option<String> {
-        Some("Android Device".to_owned())
+    fn current_monitor(app: AndroidApp) -> Self {
+        let jvm = ffi::jvm(&app);
+        let mut env = ffi::env(&jvm);
+
+        // TODO: at API level 30 use `Context.getDisplay()`.
+        // <https://developer.android.com/reference/android/content/Context#getDisplay()>
+        let id = ffi::activity(&app, &env)
+            .window_manager(&mut env)
+            .default_display(&mut env)
+            .display_id(&mut env);
+
+        Self { app, id }
     }
 
-    pub fn size(&self) -> PhysicalSize<u32> {
-        if let Some(native_window) = self.app.native_window() {
-            PhysicalSize::new(native_window.width() as _, native_window.height() as _)
-        } else {
-            PhysicalSize::new(0, 0)
-        }
+    fn available_monitors(app: AndroidApp) -> Vec<Self> {
+        let jvm = ffi::jvm(&app);
+        let mut env = ffi::env(&jvm);
+
+        let available_monitors = ffi::context(&mut env)
+            .display_manager(&mut env)
+            .displays(&mut env)
+            .into_iter()
+            .map(|display| Self { app: app.clone(), id: display.display_id(&mut env) })
+            .collect();
+
+        available_monitors
+    }
+
+    pub fn name(&self) -> Option<String> {
+        let jvm = ffi::jvm(&self.app);
+        let mut env = ffi::env(&jvm);
+
+        Some(
+            ffi::context(&mut env)
+                .display_manager(&mut env)
+                .display(&mut env, self.id)
+                .unwrap()
```

Comment:
This could panic, but is a known issue on all backends.
See https://github.com/rust-windowing/winit/issues/3258.

Reply:
Well, if you function returns `Option` you can easily avoid panic.

Reply:
I will make a follow-up PR where I finally tackle #3258.

Reply:
The thing is that monitor data, such as name, etc, is _static_ more or less. The position could change, indeed, but generally unless we make all of that into events and lifetime I'm not sure we can do much.

Reply:
Good point.
I'm happy to cache the name, but again, lets discuss it in a follow-up PR.